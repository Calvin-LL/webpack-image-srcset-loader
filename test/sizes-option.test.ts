import WISLWebpackTestCompiler from "./helpers/WISLWebpackTestCompiler";

describe.each([4, 5] as const)('v%d "sizes" option', (webpackVersion) => {
  it("should work with empty array", async () => {
    const compiler = new WISLWebpackTestCompiler({ webpackVersion });
    const bundle = await compiler.compile({
      loaderOptions: {
        sizes: [],
      },
    });

    expect(bundle.execute("main.js")).toMatchSnapshot("result");
  });

  it("should work with [null]", async () => {
    const compiler = new WISLWebpackTestCompiler({ webpackVersion });
    const bundle = await compiler.compile({
      loaderOptions: {
        sizes: [null],
      },
    });

    expect(bundle.execute("main.js")).toMatchSnapshot("result");
  });

  it('should work with ["300w"]', async () => {
    const compiler = new WISLWebpackTestCompiler({ webpackVersion });
    const bundle = await compiler.compile({
      loaderOptions: {
        sizes: ["300w"],
      },
    });

    expect(bundle.execute("main.js")).toMatchSnapshot("result");
  });

  it('should work with ["2x"]', async () => {
    const compiler = new WISLWebpackTestCompiler({ webpackVersion });
    const bundle = await compiler.compile({
      loaderOptions: {
        sizes: ["2x"],
      },
    });

    expect(bundle.execute("main.js")).toMatchSnapshot("result");
  });

  it('should work with ["1x", "2x"]', async () => {
    const compiler = new WISLWebpackTestCompiler({ webpackVersion });
    const bundle = await compiler.compile({
      loaderOptions: {
        sizes: ["1x", "2x"],
      },
    });

    expect(bundle.execute("main.js")).toMatchSnapshot("result");
  });

  it('should work with ["1x", "300w", null, "2x"]', async () => {
    const compiler = new WISLWebpackTestCompiler({ webpackVersion });
    const bundle = await compiler.compile({
      loaderOptions: {
        sizes: ["1x", "300w", null, "2x"],
      },
    });

    expect(bundle.execute("main.js")).toMatchSnapshot("result");
  });
});
