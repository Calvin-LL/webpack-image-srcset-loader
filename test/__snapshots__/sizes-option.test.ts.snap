// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"sizes" option should work with ["1x", "2x"]: result 1`] = `"44aacd9d8d250c420d100a3fb0f1debe.jpg 1x, fdaf70a107ad3cebfaf1e03ff0601c1c.jpg 2x"`;

exports[`"sizes" option should work with ["1x", "300w", null, "2x"]: result 1`] = `"44aacd9d8d250c420d100a3fb0f1debe.jpg 1x, f64d5a6cf45bcebd0bce0fd66010d2a9.jpg 300w, fdaf70a107ad3cebfaf1e03ff0601c1c.jpg, fdaf70a107ad3cebfaf1e03ff0601c1c.jpg 2x"`;

exports[`"sizes" option should work with ["2x"]: result 1`] = `"fdaf70a107ad3cebfaf1e03ff0601c1c.jpg 2x"`;

exports[`"sizes" option should work with ["300w"]: result 1`] = `"f64d5a6cf45bcebd0bce0fd66010d2a9.jpg 300w"`;

exports[`"sizes" option should work with [null]: result 1`] = `"fdaf70a107ad3cebfaf1e03ff0601c1c.jpg"`;

exports[`"sizes" option should work with empty array: result 1`] = `""`;
